kore/src/Kore/Repl.hs

```diff
@@ -6,64 +6,428 @@ License     : NCSA
+    proveSteps0 = fmap ProveSteps $ string "step" *> space *> option 1 decimal
+
+    selectNode0 :: Parser ReplCommand
+    selectNode0 =
+        fmap SelectNode $ string "select" *> space *> signed space decimal
+
+    showConfig0 :: Parser ReplCommand
+    showConfig0 = ShowConfig <$ string "config"
+
+    exit0 :: Parser ReplCommand
+    exit0 = Exit <$ string "exit"
+
+replInterpreter
+    :: forall level
+    .  MetaOrObject level
+    => ReplCommand
+    -> StateT (ReplState level) Simplifier Bool
+replInterpreter =
     \case
-        ""      -> prompt'
-        "help"  -> showHelp *> prompt'
-        "proof" -> putStrLn (unparseToString rule) *> prompt'
-        "go"    -> pure . pure $ ()
-        _       -> showError *> prompt'
+        ShowUsage -> showUsage0 $> True
+        Help -> help0 $> True
+        ShowClaim c -> showClaim0 c $> True
+        ShowAxiom a -> showAxiom0 a $> True
+        Prove i -> prove0 i $> True
+        ShowGraph -> showGraph0 $> True
+        ProveSteps n -> proveSteps0 n $> True
+        SelectNode i -> selectNode0 i $> True
+        ShowConfig -> showConfig0 $> True
+        Exit -> pure False
   where
-    prompt' = Left <$> prompt
-
-showHelp :: IO ()
-showHelp =
-    putStrLn
-        "Commands available from the prompt: \n\
-        \help                shows this help message\n\
-        \proof               shows the current proof\n\
-        \go                  starts the automated prover"
-
-showError :: IO ()
-showError = putStrLn "\nUnknown command. Try 'help'."
+    showUsage0 :: StateT st Simplifier ()
+    showUsage0 =
+        putStrLn' "Could not parse command, try using 'help'."
+
+    help0 :: StateT st Simplifier ()
+    help0 =
+        putStrLn' helpText
+
+    showClaim0 :: Int -> StateT (ReplState level) Simplifier ()
+    showClaim0 index = do
+        claim <- Lens.preuse $ lensClaims . Lens.element index
+        putStrLn' $ maybe indexNotFound (unparseToString . unClaim) claim
+
+    showAxiom0 :: Int -> StateT (ReplState level) Simplifier ()
+    showAxiom0 index = do
+        axiom <- Lens.preuse $ lensAxioms . Lens.element index
+        putStrLn' $ maybe indexNotFound (unparseToString . unAxiom) axiom
+
+    prove0 :: Int -> StateT (ReplState level) Simplifier ()
+    prove0 index = do
+        claim' <- Lens.preuse $ lensClaims . Lens.element index
+        case claim' of
+            Nothing -> putStrLn' indexNotFound
+            Just claim -> do
+                let
+                    graph@Strategy.ExecutionGraph { root }
+                        = emptyExecutionGraph claim
+                lensGraph .= graph
+                lensClaim .= claim
+                lensNode  .= root
+                putStrLn' "Execution Graph initiated"
+
+    showGraph0 :: StateT (ReplState level) Simplifier ()
+    showGraph0 = do
+        Strategy.ExecutionGraph { graph } <- Lens.use lensGraph
+
+        liftIO
+            . (flip Graph.runGraphvizCanvas') Graph.Xlib
```

Comment:
I think we could extract the combination of this line and the next line into a nice helper function. It would make this easier to read by separating the high-level of the REPL interface from the low-level of display frameworks.

Also, I see `Xlib`: does this work outside Linux?

Reply:
I'll see if I can test it on a Mac tomorrow, but I think we can consider this as an "extra" feature for demo purposes. It will definitely change in the future.